autoscale/src/main/java/com/sequenceiq/periscope/controller/DistroXAutoScaleScalingActivityV1Controller.java

```diff
@@ -0,0 +1,117 @@
+package com.sequenceiq.periscope.controller;
+
+import static java.time.temporal.ChronoUnit.MINUTES;
+
+import java.time.Instant;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByAccount;
+import com.sequenceiq.authorization.annotation.CheckPermissionByResourceCrn;
+import com.sequenceiq.authorization.annotation.CheckPermissionByResourceName;
+import com.sequenceiq.authorization.annotation.ResourceCrn;
+import com.sequenceiq.authorization.annotation.ResourceName;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.cloudbreak.auth.security.internal.TenantAwareParam;
+import com.sequenceiq.periscope.api.endpoint.v1.DistroXAutoScaleScalingActivityV1Endpoint;
+import com.sequenceiq.periscope.api.model.DistroXAutoscaleScalingActivityResponse;
+import com.sequenceiq.periscope.converter.DistroXAutoscaleScalingActivityResponseConverter;
+import com.sequenceiq.periscope.model.NameOrCrn;
+import com.sequenceiq.periscope.service.AutoscaleRestRequestThreadLocalService;
+import com.sequenceiq.periscope.service.ScalingActivityService;
+
+@Controller
+public class DistroXAutoScaleScalingActivityV1Controller implements DistroXAutoScaleScalingActivityV1Endpoint {
+
+    public static final Integer MINUTES_TO_SUBTRACT = 60;
+
+    @Inject
+    private DistroXAutoscaleScalingActivityResponseConverter distroXAutoscaleScalingActivityResponseConverter;
+
+    @Inject
+    private ScalingActivityService scalingActivityService;
+
+    @Inject
+    private AutoscaleRestRequestThreadLocalService threadLocalService;
+
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.DESCRIBE_DATAHUB)
```

Comment:
DESCRIBE_DATAHUB check works on resource (datahub) level, so `CheckPermissionByAccount` won't work in mow envs

Reply:
@horadla23  - this should be CheckPermissionByResourceCrn ?

Reply:
> Or does this need to be re-structured to have a crn, but ignore while looking up the operationId if it is not needed.

from authz point of view it does not make sense. then you could still query all operationId independently from the cluster.
I would introduce a clusterName/clusterCrn param and filter for scaling activities based on the cluster.
The whole point of authz is to enable calling the endpoint only if user has permission over related resource.
So endpoint should look like this:
```
@Override
@CheckPermissionByResourceCrn(action = AuthorizationResourceAction.DESCRIBE_DATAHUB)
public DistroXAutoscaleScalingActivityResponse getScalingActivityUsingOperationId(@ResourceCrn @TenantAwareParam String clusterCrn, String operationId) {
        String accountId = threadLocalService.getCloudbreakTenant();
        return distroXAutoscaleScalingActivityResponseConverter.convert(scalingActivityService.findByOperationIdAndClusterCrn(operationId, clusterCrn));
    }
```

cc @Vatsal23082000 I think we already agreed on similar solution

Reply:
Done